test/unit/buffer_spec.lua

```diff
@@ -212,92 +213,148 @@ describe('buffer functions', function()
 
-      local width = build_stl_str_hl("%n")
+        local actual_width = build_stl_str_hl(stl, input_width)
 
-      eq(1, width)
-      eq("1", helpers.ffi.string(output_buffer, width))
-    end)
-
-    it('should print the current line number in the buffer', function()
-      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
-
-      local width = build_stl_str_hl("%l")
-
-      eq(1, width)
-      eq("0", helpers.ffi.string(output_buffer, width))
-
-    end)
-
-    it('should print the number of lines in the buffer', function()
-      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
-
-      local width = build_stl_str_hl("%L")
-
-      eq(1, width)
-      eq("1", helpers.ffi.string(output_buffer, width))
+        eq(actual_width, expected_width)
+        eq(expected_stl, get_str(output_buffer, expected_width))
+      end)
+    end
 
-    end)
+    -- TODO: Add tests for many `=` markers
+    --
+
+    statusline_test('should copy plain text', 80, 14, 'this is a test', 'this is a test')
+    statusline_test('should print the tail file name', 80, 15, '%t', 'buffer_spec.lua')
+    statusline_test('should print the buffer number', 80, 1, '%n', '1')
+    statusline_test('should print the current line number in the buffer', 80, 1, '%l', '0')
+    statusline_test('should print the number of lines in the buffer', 80, 1, '%L', '1')
+
+    statusline_test('should truncate when standard text pattern is too long', 10, 10,
+      '0123456789abcde',     '<6789abcde')
+    statusline_test('should truncate when using =', 10, 10,
+      'abcdef%=ghijkl',      'abcdef<jkl')
+    statusline_test('should truncate centered text when using ==', 10, 10,
+      'abcde%=gone%=fghij',  'abcde<ghij')
+
+    statusline_test('should right align when using =', 20, 20,
+      'neo%=vim',            'neo              vim')
+    statusline_test('should, when possible, center text when using %=text%=', 20, 20,
+      'abc%=neovim%=def',    'abc    neovim    def')
+    statusline_test('should completely fill the buffer when using %=text%=', 20, 20,
+      'abc%=neo_vim%=def',   'abc   neo_vim    def')
+    statusline_test('should have equal spaces even with non-equal sides when using =', 20, 20,
+      'foobar%=test%=baz',   'foobar   test    baz')
+    statusline_test('should have equal spaces even with longer right side when using =', 20, 20,
+      'a%=test%=longtext',   'a   test    longtext')
+    statusline_test('should handle an empty left side when using ==', 20, 20,
+      '%=test%=baz',         '      test       baz')
+    statusline_test('should handle an empty right side when using ==', 20, 20,
+      'foobar%=test%=',      'foobar     test     ')
+
+    statusline_test('should approximately center text when using %=text%=', 21, 21,
+      'abc%=neovim%=def',    'abc    neovim     def')
+    statusline_test('should completely fill the buffer when using %=text%=', 21, 21,
```

Comment:
Test name is rather strange: buffer is always filled completely, is not it?

(Also still where is `%=text%=` test (no text on both sides).) And I can suggest

- `%=%=text%=` (three empty sections: test regarding how `%=%=` is handled: situation may appear when some of the center sections appears empty).
- `%=text`, `text%=`
- `%=` and just empty string: such edge cases are more likely to provoke a crash.
- `('%='):rep( --[[STL_MAX_ITEM]] 80)` (probably use `local STL_MAX_ITEM = 80`)
- `('%='):rep( --[[STL_MAX_ITEM + 1]] 81)` (so here it would be `STL_MAX_ITEM + 1`)